src/components/modals/ImportAccounts/index.js

```diff
@@ -0,0 +1,224 @@
+
+class ImportAccounts extends PureComponent<Props, State> {
+  state = INITIAL_STATE
+  STEPS = createSteps({
+    t: this.props.t,
+  })
+
+  transitionTo = stepId => {
+    let nextState = { stepId }
+    if (stepId === 'chooseCurrency') {
+      nextState = { ...INITIAL_STATE }
+    }
+    this.setState(nextState)
+  }
+
+  handleClickImport = async () => {
+    const { addAccount } = this.props
+    const { scannedAccounts, checkedAccountsIds } = this.state
+    const accountsIdsMap = checkedAccountsIds.reduce((acc, cur) => {
+      acc[cur] = true
+      return acc
+    }, {})
+    const accountsToImport = scannedAccounts.filter(account => accountsIdsMap[account.id] === true)
+    for (let i = 0; i < accountsToImport.length; i++) {
+      await idleCallback()
+      addAccount(accountsToImport[i])
+    }
+    this.transitionTo('finish')
+  }
+
+  handleCloseModal = () => {
+    const { closeModal } = this.props
+    closeModal('importAccounts')
+  }
+
+  render() {
+    const { t, currentDevice, existingAccounts } = this.props
+    const {
+      stepId,
+      currency,
+      isAppOpened,
+      scannedAccounts,
+      checkedAccountsIds,
+      scanStatus,
+      err,
+    } = this.state
+
+    const stepIndex = this.STEPS.findIndex(s => s.id === stepId)
+    const step = this.STEPS[stepIndex]
+
+    if (!step) {
+      throw new Error(`ImportAccountsModal: step ${stepId} doesn't exists`)
+    }
+
+    const { component: StepComponent, footer: StepFooter, hideFooter, onBack } = step
+
+    const stepProps: StepProps = {
+      t,
+      currency,
+      currentDevice,
+      existingAccounts,
+      scannedAccounts,
+      checkedAccountsIds,
+      scanStatus,
+      err,
+      isAppOpened,
+      onClickImport: this.handleClickImport,
+      onCloseModal: this.handleCloseModal,
+      transitionTo: this.transitionTo,
+      setState: (...args) => this.setState(...args),
+    }
+
+    return (
+      <Modal
+        name="importAccounts"
+        preventBackdropClick
+        onHide={() => this.setState({ ...INITIAL_STATE })}
+        render={({ onClose }) => (
+          <ModalBody onClose={onClose}>
+            <ModalTitle onBack={onBack ? () => onBack(stepProps) : void 0}>
+              {t('importAccounts:title')}
+            </ModalTitle>
+            <ModalContent>
+              <Breadcrumb mb={6} currentStep={stepIndex} items={this.STEPS} />
+              <StepComponent {...stepProps} />
+            </ModalContent>
+            {!hideFooter && (
+              <ModalFooter horizontal align="center" justify="flex-end" style={{ height: 80 }}>
+                {StepFooter ? <StepFooter {...stepProps} /> : <Box>footer</Box>}
+              </ModalFooter>
+            )}
+          </ModalBody>
+        )}
+      />
+    )
+  }
+}
+
+export default compose(connect(mapStateToProps, mapDispatchToProps), translate())(ImportAccounts)
+
+function idleCallback() {
+  return new Promise(resolve => window.requestIdleCallback(resolve))
```

Comment:
I'm pretty sure it's an antipattern and is equivalent to `Promise.resolve()` . i used to do same with requestAnimationFrame but fact is, a promise .then runs in the next tick, so either directly use requestIdleCallback but probably don't transform it to a promise

Reply:
ultimately, should we instead have a addAccounts action?